library/ord-tree.lisp

```diff
@@ -125,7 +125,9 @@
        (match (<=> needle elt)
          ((LT) (lookup left needle))
          ((EQ) (Some elt))
-         ((GT) (lookup right needle))))))
+         ((GT) (lookup right needle))))
+      (_
```

Comment:
I would replace this catch-all clause with `((DoubleBlackEmpty) (error "Found double-black node outside of removal process"))`. One of the invariants of the data structure is that double-black nodes only exist within the removal algorithm, and are cleaned up before a `remove` (or similar) returns.